ra2ce/analysis/adaptation/adaptation_option_analysis.py

```diff
@@ -0,0 +1,137 @@
+        """
+        Get the analysis class and the result column for the given analysis.
+
+        Args:
+            analysis (AnalysisDamagesEnum | AnalysisLossesEnum): The type of analysis.
+
+        Raises:
+            NotImplementedError: The analysis type is not implemented.
+
+        Returns:
+            tuple[type[Damages | LossesBase], str]: The analysis class and the result column.
+        """
+        if analysis_type == AnalysisDamagesEnum.DAMAGES:
+            return (Damages, "dam_")
+        elif analysis_type == AnalysisLossesEnum.SINGLE_LINK_LOSSES:
+            return (SingleLinkLosses, "vlh_.*_total")
+        elif analysis_type == AnalysisLossesEnum.MULTI_LINK_LOSSES:
+            return (MultiLinkLosses, "vlh_.*_total")
+        raise NotImplementedError(f"Analysis {analysis_type} not implemented")
+
+    @classmethod
+    def from_config(
+        cls,
+        analysis_config: AnalysisConfigWrapper,
+        analysis_type: AnalysisLossesEnum | AnalysisDamagesEnum,
+        option_id: str,
+    ) -> AdaptationOptionAnalysis:
+        """
+        Classmethod to create an AdaptationOptionAnalysis from an analysis configuration.
+
+        Args:
+            analysis_config (AnalysisConfigWrapper): The analysis configuration.
+            analysis_type (AnalysisLossesEnum | AnalysisDamagesEnum): The type of analysis.
+            option_id (str): The ID of the adaptation option.
+
+        Returns:
+            AdaptationOptionAnalysis: The created AdaptationOptionAnalysis.
+        """
+
+        # Need a deepcopy to avoid mixing up configs across analyses.
+        _analysis_config = deepcopy(analysis_config)
+        _analysis_config.config_data = (
+            _analysis_config.config_data.reroot_analysis_config(
+                analysis_type,
+                analysis_config.config_data.root_path.joinpath("input", option_id),
+            )
+        )
+
+        # Create analysis input
+        _analysis = _analysis_config.config_data.get_analysis(analysis_type)
+        if analysis_type == AnalysisDamagesEnum.DAMAGES:
+            _graph_file = None
+            _graph_file_hazard = analysis_config.graph_files.base_network_hazard
+        else:
+            _graph_file = analysis_config.graph_files.base_graph
+            _graph_file_hazard = analysis_config.graph_files.base_graph_hazard
+
+        _analysis_input = AnalysisInputWrapper.from_input(
+            analysis=_analysis,
+            analysis_config=_analysis_config,
+            graph_file=_graph_file,
+            graph_file_hazard=_graph_file_hazard,
+        )
+
+        # Create output object
+        _analysis_class, _result_col = cls.get_analysis(analysis_type)
+
+        return cls(
+            analysis_class=_analysis_class,
+            analysis_input=_analysis_input,
+            result_col=_result_col,
+        )
+
+    def execute(self, analysis_config: AnalysisConfigWrapper) -> GeoDataFrame:
```

Comment:
Is this the executo of the `AnalysisProtocol`? If so then you are missing the protocol declaration at the dataclass level.

Reply:
The execute you could indeed see as implementation of the protocol. However, the class doesn't have/need all the properties of the AnalysisProtocol. So I propose not to implement it officially.